lein-finagle-clojure/src/leiningen/finagle_clojure.clj

```diff
@@ -33,15 +55,29 @@
     lein finagle-clojure scrooge :lint --help # shows available options for the linter
     lein finagle-clojure scrooge :lint -w # show linter warnings as well (warnings won't prevent compilation)"
   [project & options]
+  (doseq [f (classpath/get-classpath project)]
+    (pom/add-classpath f))
   (let [subtask (first options)
         project-root (:root project)
         source-path (get-in project [:finagle-clojure :thrift-source-path])
         raw-dest-path (get-in project [:finagle-clojure :thrift-output-path])]
     (if-not (and source-path raw-dest-path)
       (leiningen.core.main/info "No config found for lein-finagle-clojure, not compiling Thrift for" (:name project))
       (let [absolute-dest-path (->> raw-dest-path (io/file project-root) (.getAbsolutePath))
-            thrift-files (thrift-files project-root source-path)
-            scrooge-args (concat ["--finagle" "--skip-unchanged" "--language" "java" "--dest" absolute-dest-path] thrift-files)]
+            thrift-files (find-thrift-files project-root source-path)
+            include-filenames (->> thrift-files
+                                   (mapcat #(scrape-includes (slurp %)))
+                                   ;; Verify dependency file isn't already in the thrift source path
+                                   (filter #(every? (fn [^String thrift-file]
+                                                      (not (.endsWith thrift-file %)))
+                                                    thrift-files)))
+            _ (doseq [filename include-filenames
```

Comment:
I use `_` in `let`s a lot around my Clojure code. I see it as a relatively harmless idiom. I agree the `doseq` technically doesn't need to be in the let here. I figured this positioning allows for the best flow when reading through the code. It's more obvious that I'm copying files into the temp directory, then pointing scrooge at that directory. But I'm flexible if you feel strongly about it.